### ISSUE TITLE
(suggestion) kill the smallest plot when memory is full

### ISSUE THREAD

**Issue by dJani97** [swar/Swar-Chia-Plot-Manager#643]
Hi!

I'm very surprised by the amount of improvements in the latest version. Really, really awesome work, thank you!

I have a suggestion, if you'd happen to run out of them (which I doubt) 😁

I'm trying to max out my plotting capacity and I often end up overallocating my memory. This results in a bunch of plots repeatedly running out of memory and dying. They also leave behing some garbage, eventually clogging up most temp drives.

It would be a "nice to have" feature to be able and set a memory tresshold, eg. 95%, and when Plot Manager detects that the system went above this treshold, it would kill the smallest plot (with lowest percentage) and remove it's TEMP files.

This would ensure a much greater stability even when the configuration is not perfect.

In the meantime to warn the user about such events, the View could display something like "Plots killed due to insufficient memory: 1" when this happens. And maybe even send a notification.

As a workaround, [do you know any good (and free) monitoring programs](https://www.reddit.com/r/ChiaMining/comments/nh3zz3/monitor_chia_rig_cpu_and_ram_usage/) that could notify me in any way when memory usage is critically high? If you do, let me know here or on Reddit.

**Comment by asukahan** [comment]
RAM usage is a static value, it's VERY EASY to manage your RAM usage of plot missions. please do a simple calculation (RAM x Number of missions) before starting your plot mission.

**Comment by dJani97** [comment]
It should be, right?

I have a plotting PC with 32GBs of RAM.

In my current config, I have three jobs. Taking into account even the "early start" part, which should be a pretty rare scenario, the max memory usage of plotting should be 26440 MB.

It still runs out of memory once in a while.
